src/app/qgisapp.cpp

```diff
@@ -9955,26 +9975,68 @@ void QgisApp::new3DMapCanvas()
     return;
   }
 
-  Qgs3DMapSettings *map = new Qgs3DMapSettings;
-  map->setCrs( prj->crs() );
-  map->setOrigin( fullExtent.center().x(), fullExtent.center().y(), 0 );
-  map->setSelectionColor( mMapCanvas->selectionColor() );
-  map->setBackgroundColor( mMapCanvas->canvasColor() );
-  map->setLayers( mMapCanvas->layers() );
+  int i = 1;
 
-  QgsFlatTerrainGenerator *flatTerrain = new QgsFlatTerrainGenerator;
-  flatTerrain->setCrs( map->crs() );
-  flatTerrain->setExtent( fullExtent );
-  map->setTerrainGenerator( flatTerrain );
+  bool existing = true;
+  const QList< Qgs3DMapCanvas * > existingCanvases = findChildren< Qgs3DMapCanvas * >();
+  QString name;
+  while ( existing )
+  {
+    name = tr( "3D Map %1" ).arg( i++ );
```

Comment:
Random thought ... But is "3d map" the right term here? Should it be "3d view" instead? "3d scene"?

Reply:
I do not really have any preference here - we can use whatever sounds most natural to native speakers.

Hopefully my favorite "Ultimate 3D Map View Pro (TM)" can be considered as well when choosing the right name.

Reply:
@nyalldawson , @wonder-sk , while I have no strong feeling here, I do like 3D Map [View], as it sits quite nicely next to the Map [View]:
![screenshot from 2017-10-19 10-09-57](https://user-images.githubusercontent.com/1728657/31752583-dd05e674-b4b5-11e7-9041-cdccc70e1089.png)

In any case, let's merge this 😉